test/rules/noUnusedVariableRuleTests.ts

```diff
@@ -113,4 +113,70 @@ describe("<no-unused-variable>", () => {
 
         assert.lengthOf(actualFailures, 0);
     });
+
+    describe("react option set to", () => {
```

Comment:
these describe/it blocks could use a little refactoring now that there is only one possible react option. I think the cases should be something like:
React Option Set:
  * No error if JSX present
  * No error if React. usage present
  * error if neither is present
React Option Not Set:
  * Failure if no usage of React
  * Failure if only JSX
  * No failure if React. is present

(and then also a couple similar tests for react/addons)